storage/operation/pebbleimpl/writer.go

```diff
@@ -66,11 +66,10 @@ func (b *ReaderBatchWriter) AddCallback(callback func(error)) {
 }
 
 // Commit flushes the batch to the database.
-// No errors are expected during normal operation.
+// Commit may be called at most once per Batch.
 // ReaderBatchWriter can't be reused after Commit() is called.
+// No errors are expected during normal operation.
 func (b *ReaderBatchWriter) Commit() error {
-	defer b.batch.Close() // Release batch resource
```

Comment:
Thanks Jordon. Good catch 👍 . 

Just that when `Commit` alone is called without `WithReaderBatchWriter` (which calls Commit and Close internally),  we need to make sure `Close` is called after `Commit`.

There are some places where `Commit` is called alone, [the block iterator is one of it](https://github.com/onflow/flow-go/blob/master/module/block_iterator/executor/executor.go#L87), I will work on a refactor to add the call to the Close method.

Reply:
https://github.com/onflow/flow-go/pull/7490